rdkit/Chem/EState/EState_VSA.py

```diff
@@ -71,48 +75,59 @@ def EState_VSA_(mol, bins=None, force=1):
   ans = numpy.zeros(len(bins) + 1, numpy.float)
   for i, prop in enumerate(propContribs):
     if prop is not None:
-      bin = bisect.bisect_right(bins, prop)
-      ans[bin] += volContribs[i + 1]
+      nbin = bisect.bisect_right(bins, prop)
+      ans[nbin] += volContribs[i + 1]
   mol._eStateVSA = ans
   return ans
 
 
+def _descriptorDocstring(name, nbin, bins):
+  """ Create a docstring for the descriptor name """
+  if nbin == 0:
+    interval = "-inf < x <  {0:.2f}".format(bins[nbin])
+  elif nbin < len(bins):
+    interval = " {0:.2f} <= x <  {1:.2f}".format(bins[nbin - 1], bins[nbin])
+  else:
+    interval = " {0:.2f} <= x < inf".format(bins[nbin - 1])
+  return '{0} Descriptor {1} ({2})'.format(name, nbin + 1, interval)
+
+
+def _descriptor_VSA_EState(nbin):
+
+  def VSA_EState_bin(mol):
+    return VSA_EState_(mol, force=False)[nbin]
+
+  name = "VSA_EState{0}".format(nbin + 1)
+  fn = VSA_EState_bin
+  fn.__doc__ = _descriptorDocstring('VSA EState', nbin, vsaBins)
+  fn.version = '1.0.0'
+  return name, fn
+
+
+def _descriptor_EState_VSA(nbin):
+
+  def EState_VSA_bin(mol):
+    return EState_VSA_(mol, force=False)[nbin]
+
+  name = "EState_VSA{0}".format(nbin + 1)
+  fn = EState_VSA_bin
+  fn.__name__ = name
+  if hasattr(fn, '__qualname__'):
+    fn.__qualname__ = name
+  fn.__doc__ = _descriptorDocstring('EState VSA', nbin, estateBins)
+  fn.version = '1.0.1'
+  return name, fn
+
+
 def _InstallDescriptors():
```

Comment:
I found this code very hard to understand and made the function declaration more explicit.